dns_test.go

```diff
@@ -374,17 +374,47 @@ func TestDNSProtocol(t *testing.T) {
 
 		_, port, _ := net.SplitHostPort(addr.String())
 
-		// Force IPv4
+		// Default DNS Server port
 		module := Module{
 			Timeout: time.Second,
 			DNS: DNSProbe{
 				QueryName: "example.com",
-				Protocol:  protocol + "4",
+				Protocol:  protocol,
 			},
 		}
 		recorder := httptest.NewRecorder()
 		registry := prometheus.NewRegistry()
-		result := probeDNS(net.JoinHostPort("localhost", port), recorder, module, registry)
+		result := probeDNS("localhost", recorder, module, registry)
+		if result {
+			t.Fatalf("DNS protocol: \"%v\" with localhost:53 succeeded, expected failure.", protocol)
```

Comment:
I'm sorry, I don't understand that, as the [test assume](https://github.com/Colstuwjx/blackbox_exporter/blob/5fc97cca1dae3824b1a598b122d11a7bdd38d891/dns_test.go#L361) said that `localhost` resolves both `127.0.0.1` & `::1`